EdFi.Ods.Utilities.Migration/Providers/MigrationConfigurationProvider.cs

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Linq;
+using EdFi.Ods.Utilities.Migration.Configuration;
+using EdFi.Ods.Utilities.Migration.Enumerations;
+using EdFi.Ods.Utilities.Migration.Helpers;
+using log4net;
+
+namespace EdFi.Ods.Utilities.Migration.Providers
+{
+    public class MigrationConfigurationProvider : IMigrationConfigurationProvider
+    {
+        private readonly IOdsMigrationManagerResolver _odsMigrationManagerResolver;
+        private readonly ICurrentOdsApiVersionProvider _currentOdsApiVersionProvider;
+        private readonly ILog _logger = LogManager.GetLogger(typeof(MigrationConfigurationProvider));
+
+        public MigrationConfigurationProvider(IOdsMigrationManagerResolver odsMigrationManagerResolver, ICurrentOdsApiVersionProvider currentOdsApiVersionProvider)
+        {
+            _odsMigrationManagerResolver = odsMigrationManagerResolver;
+            _currentOdsApiVersionProvider = currentOdsApiVersionProvider;
+        }
+
+        public UpgradeVersionConfiguration Get(Options options,
+            string optionalCurrentOdsVersionOverride = null,
+            string optionalRequestedUpgradeVersionOverride = null)
+{
+            var upgradeVersionConfiguration = new UpgradeVersionConfiguration();
+
+            if (string.IsNullOrEmpty(optionalCurrentOdsVersionOverride))
```

Comment:
As mentioned, even if we're overriding the version, we still have to detect features from the connection string.

Reply:
done